### PR TITLE
zellij: 0.40.1 -> 0.41.1

### PR DESCRIPTION
**Pull request by r-vdp**
https://github.com/zellij-org/zellij/releases/tag/v0.41.0

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by jorikvanveen** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353630`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>zellij</li>
  </ul>
</details>

**Comment by a-kenji** [comment]
Maintainers should be aware of the following:

> Zellij `0.41.0` forces vendored libcurl dependency by default

Source:
https://github.com/zellij-org/zellij/issues/3720

**Comment by thegrubster** [comment]
There has been a new patch release due to some issues with the generic binaries in `0.41`, although I am unsure if this is relevant for nix

https://github.com/zellij-org/zellij/releases/tag/v0.41.1

**Comment by a-kenji** [comment]
Thank you @thegrubster,

> There has been a new patch release due to some issues with the generic binaries in 0.41, although I am unsure if this is relevant for nix

Yes, this seems relevant.

**Comment by tjni** [comment]
Maybe we can patch `zellij-utils/Cargo.toml` to depend on `isahc` without the static-curl feature.

**Review by nadir-ishiguro — APPROVED** [review]

**Comment by nadir-ishiguro** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353630`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>zellij</li>
  </ul>
</details>

**Review by wahjava — APPROVED** [review]
Looks good on `aarch64-linux`.

**Comment by r-vdp** [comment]
> Maybe we can patch `zellij-utils/Cargo.toml` to depend on `isahc` without the static-curl feature.

I actually tried this, but it doesn't seem to work. I added
```nix
  postPatch = ''
    substituteInPlace zellij-utils/Cargo.toml \
      --replace-fail 'isahc = "1.7.2"' 'isahc = { version = "1.7.2", default-features = false, features = ["http2", "text-decoding"] }'
  '';
```

But I'm still not seeing curl as a dependency:
```bash
➜ nix-store --query --references result/
/nix/store/3bvxjkkmwlymr0fssczhgi39c3aj1l7i-glibc-2.40-36
/nix/store/m8gwqmn8k3jm0gbcia358mz4y00lgmbc-openssl-3.3.2
/nix/store/s94fwp43xhzkvw8l8nqslskib99yifzi-gcc-13.3.0-lib

➜ ldd ./result/bin/zellij
        linux-vdso.so.1 (0x00007f81611ab000)
        libssl.so.3 => /nix/store/m8gwqmn8k3jm0gbcia358mz4y00lgmbc-openssl-3.3.2/lib/libssl.so.3 (0x00007f815f2f4000)
        libcrypto.so.3 => /nix/store/m8gwqmn8k3jm0gbcia358mz4y00lgmbc-openssl-3.3.2/lib/libcrypto.so.3 (0x00007f815ec00000)
        libc.so.6 => /nix/store/3bvxjkkmwlymr0fssczhgi39c3aj1l7i-glibc-2.40-36/lib/libc.so.6 (0x00007f815ea07000)
        libgcc_s.so.1 => /nix/store/s94fwp43xhzkvw8l8nqslskib99yifzi-gcc-13.3.0-lib/lib/libgcc_s.so.1 (0x00007f8161180000)
        libm.so.6 => /nix/store/3bvxjkkmwlymr0fssczhgi39c3aj1l7i-glibc-2.40-36/lib/libm.so.6 (0x00007f815f20d000)
        libdl.so.2 => /nix/store/3bvxjkkmwlymr0fssczhgi39c3aj1l7i-glibc-2.40-36/lib/libdl.so.2 (0x00007f815f206000)
        libpthread.so.0 => /nix/store/3bvxjkkmwlymr0fssczhgi39c3aj1l7i-glibc-2.40-36/lib/libpthread.so.0 (0x00007f815f201000)
        /nix/store/3bvxjkkmwlymr0fssczhgi39c3aj1l7i-glibc-2.40-36/lib/ld-linux-x86-64.so.2 => /nix/store/3bvxjkkmwlymr0fssczhgi39c3aj1l7i-glibc-2.40-36/lib64/ld-linux-x86-64.so.2 (0x00007f81611ad000)
```

Am I holding this wrong?

**Comment by tjni** [comment]
Does it need `curl` in `buildInputs`?

**Comment by tjni** [comment]
I verified that this is needed on darwin at least (due to https://github.com/zellij-org/zellij/pull/3668), but I'm having some trouble building this on aarch64-linux through the darwin linux builder (I don't have access to a Linux machine at the moment).

**Comment by r-vdp** [comment]
> Does it need `curl` in `buildInputs`?

Yeah, I didn't mention it but I also added curl to the build inputs of course :)

**Comment by r-vdp** [comment]
@tjni I also found that by looking what other packages in nixpkgs are doing, it's working now!

**Comment by r-vdp** [comment]
It would be helpful if anyone could test this on darwin also.

File: pkgs/tools/misc/zellij/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/misc/zellij/default.nix]: For curl-config, right?

Comment (r-vdp) [reply pkgs/tools/misc/zellij/default.nix]: So that cargo finds curl, using pkg-config. Without this, it doesn't.

File: pkgs/by-name/ze/zellij/package.nix
Comment (tjni) [inline pkgs/by-name/ze/zellij/package.nix:55]: I tested earlier that these can be entirely removed after https://github.com/NixOS/nixpkgs/pull/346043. I mention it now because since you are moving this package to by-name, removing these inputs also means you can remove the entry entirely from all-packages.nix.

File: pkgs/by-name/ze/zellij/package.nix
Comment (tjni) [inline pkgs/by-name/ze/zellij/package.nix:43]: I think we can go with this for now, but I think cross-compilation might be broken.

1. curl-sys uses https://docs.rs/pkg-config/latest/pkg_config/#cross-compilation, which seems to require setting certain environment variables for cross-compilation use case.
2. we need it to use at build-time the .pc files from curl used at runtime, so how do we do that?

I don't have enough experience in cross-compilation to know how to do this. Maybe @Artturin or someone else can help out here (perhaps as a follow up to this PR)?

Note: This is technically not needed on Darwin since pkg-config is not used on Darwin.

File: pkgs/by-name/ze/zellij/package.nix
Comment (r-vdp) [reply pkgs/by-name/ze/zellij/package.nix:55]: Ah right, ok, I'll do that!

File: pkgs/by-name/ze/zellij/package.nix
Comment (r-vdp) [reply pkgs/by-name/ze/zellij/package.nix:43]: Would you make its inclusion conditional here, or do we consider that it doesn't matter to include this path in the sandbox also on darwin and it will simply not be used then?

For the cross-compilation, I also don't have much experience with this.

Comment (tjni) [reply pkgs/by-name/ze/zellij/package.nix:43]: I've been thinking about it a little more, and I think we should just keep it like this for now. There's probably a nice solution for this somewhere, but I'll mull it over some more separately.

**Comment by tjni** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353630`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>zellij</li>
  </ul>
</details>

**Review by tjni — APPROVED** [review]
LGTM for Darwin

**Comment by khaneliman** [comment]
Does 0.41.1 work without 21de802c9f23bb75a02a737aaf928200c10fba3d and did we not need it before the version bump? Otherwise that should probably be squashed into the first commit if it's unique to the version bump.

**Comment by r-vdp** [comment]
> Does 0.41.1 work without [21de802](https://github.com/NixOS/nixpkgs/commit/21de802c9f23bb75a02a737aaf928200c10fba3d) and did we not need it before the version bump? Otherwise that should probably be squashed into the first commit if it's unique to the version bump to avoid a commit that technically is breaking a package (Doesn't truly matter, just a nit to think about for atomic commits.). Otherwise, LGTM.

It does work, it just has a vendored curl binary, which is not ideal in terms of security updates.
I'd like to keep the commits separate, since they are kind of orthogonal concerns. But I also don't feel very strong either way.

**Comment by khaneliman** [comment]
> > Does 0.41.1 work without [21de802](https://github.com/NixOS/nixpkgs/commit/21de802c9f23bb75a02a737aaf928200c10fba3d) and did we not need it before the version bump? Otherwise that should probably be squashed into the first commit if it's unique to the version bump to avoid a commit that technically is breaking a package (Doesn't truly matter, just a nit to think about for atomic commits.). Otherwise, LGTM.
> 
> It does work, it just has a vendored curl binary, which is not ideal in terms of security updates. I'd like to keep the commits separate, since they are kind of orthogonal concerns. But I also don't feel very strongly either way.

If it works without it, that's fine to me.

**Review by khaneliman — APPROVED** [review]

**Pull request merged**
